examples/miscellaneous/miscellaneous_ex16/miscellaneous_ex16.C

```diff
@@ -0,0 +1,496 @@
+      // side MUST live on a boundary of the domain.
+      for (auto side : elem->side_index_range())
+        if (elem->neighbor_ptr(side) == nullptr)
+        {
+          // The value of the shape functions at the quadrature
+          // points.
+          const std::vector<std::vector<Real>> & phi_face = fe_face->get_phi();
+
+          // The Jacobian * Quadrature Weight at the quadrature
+          // points on the face.
+          const std::vector<Real> & JxW_face = fe_face->get_JxW();
+
+          // The XYZ locations (in physical space) of the
+          // quadrature points on the face.  This is where
+          // we will interpolate the boundary value function.
+          const std::vector<Point> & qface_point = fe_face->get_xyz();
+
+          // Compute the shape function values on the element
+          // face.
+          fe_face->reinit(elem, side);
+
+          // Some shape functions will be 0 on the face, but for
+          // ease of indexing and generality of code we loop over
+          // them anyway
+          libmesh_assert_equal_to(n_dofs, phi_face.size());
+
+          // Loop over the face quadrature points for integration.
+          for (unsigned int qp = 0; qp < qface.n_points(); qp++)
+          {
+            // The location on the boundary of the current
+            // face quadrature point.
+            const Real xf = qface_point[qp](0);
+            const Real yf = qface_point[qp](1);
+
+            // The penalty value.  \frac{1}{\epsilon}
+            // in the discussion above.
+            const Real penalty = 1.e10;
+
+            // The boundary value.
+            const Real value = exact_solution(xf, yf);
+
+            // Matrix contribution of the L2 projection.
+            for (unsigned int i = 0; i != n_dofs; i++)
+              for (unsigned int j = 0; j != n_dofs; j++)
+                Ke(i, j) += JxW_face[qp] * penalty * phi_face[i][qp] * phi_face[j][qp];
+
+            // Right-hand-side contribution of the L2
+            // projection.
+            for (unsigned int i = 0; i != n_dofs; i++)
+              Fe(i) += JxW_face[qp] * penalty * value * phi_face[i][qp];
+          }
+        }
+    }
+
+    // We have now finished the quadrature point loop,
+    // and have therefore applied all the boundary conditions.
+
+    // If this assembly program were to be used on an adaptive mesh,
+    // we would have to apply any hanging node constraint equations
+    dof_map.constrain_element_matrix_and_vector(Ke, Fe, dof_indices);
+
+    if (sc)
+      sc->set_current_elem(*elem);
+
+    // The element matrix and right-hand-side are now built
+    // for this element.  Add them to the global matrix and
+    // right-hand-side vector.  The  SparseMatrix::add_matrix()
+    // and  NumericVector::add_vector() members do this for us.
+    matrix.add_matrix(Ke, dof_indices);
+    system.rhs->add_vector(Fe, dof_indices);
+  }
+
+  matrix.close();
+}
+
+#else
+
+int
+main()
+{
+  return 0;
```

Comment:
Could we instead get a `libmesh_example_requires()` for the case where we need a different configuration?